### DISCUSSION TITLE
Ansys mesh file input to moose

### DISCUSSION THREAD

**Discussion by nishant1785** [idaholab/moose#22531] [Q&A Modules: General] [answered]
Hi,
For first time I have tried using ansys mesh (.msh format) file as an input to a simple diffusion code.  But i get following error in moose:
```
*** ERROR ***
Invalid dimension dim = 4
```
Note:Actually geometry was generated in abaqus and then imported in ansys to make *.msh mesh file which was later used in MOOSE as mentioned above.

Thank you.

**Comment by GiudGiud** [comment]
@jortensi @permcody do you know anyone who has been doing ANSYS->ABAQUS->MOOSE ?

**Reply by permcody** [reply]
I don't believe we've ever tried to read Ansys meshes. We do have an Abaqus reader though, but it may not be full featured. libMesh (our I/O library) supports many formats, gmsh, exodus, and even Ascii formats (XDA). You might also consider using our  MeshGenerator system, where you can programmatically build up parameterizeable meshes.

**Comment by nishant1785** [comment]
Okk thank you I will see that.

Nishant Gaikwad-7972836355
M. Tech student
Materials Department
DIAT, DRDO,Girinagar
411025-Pune

**Comment by Wang-Yihu** [comment, marked as answer]
Haha, MOOSE Multiphysics cannot read ANSYS mesh (*.msh) in the current version.
I suggest you: 
1. export ANSYS Workbench mesh in FLUENT input mesh (*.msh)
2. read the *.msh with ICEM CFD
3. set the Exodus format, set the boundary conditions, and export it (*.exo)
This really works!!!! I use this method!!!!!!
